lib/Traits/ShadowTraits.ts

```diff
@@ -1,9 +1,9 @@
+import CatalogMemberTraits from "./CatalogMemberTraits";
 import primitiveTrait from "./primitiveTrait";
-import ModelTraits from "./ModelTraits";
 
 export type Shadows = "CAST" | "RECEIVE" | "BOTH" | "NONE";
 
-export default class ShadowTraits extends ModelTraits {
+export default class ShadowTraits extends CatalogMemberTraits {
```

Comment:
This seems kind of lazy to me. Something with ShadowTraits is not inherently a catalog member. It might be one, in fact it almost certainly is, but it's a coincidence rather than something that should be codified into the inheritance tree, IMO.

Reply:
Can you split out the traits from CatalogMemberTraits that you want to use here and share it across CatalogMember and ShadowTraits?

Reply:
I agree, I don't really like it. But I have done it this way as I need to access `super.selectableDimensions` in `ShadowMixin` to merge them:

https://github.com/TerriaJS/terriajs/blob/e5546b6cca4a3bb8200c1095c3487d1e5cff13ca/lib/ModelMixins/ShadowMixin.ts#L33-L53

We don't currently have a mechanism to merge arrays across Mixins (similar to how we deal with arrays across strata). So there isn't really a nice way to merge all of the `selectableDimensions` across all of the different Mixins.

Reply:
Actually I'll just rename `ShadowMixin.selectableDimensions` to `ShadowMixin.shadowDimensions` or something and then merge them upstream

Reply:
Done